pandas/core/indexes/datetimelike.py

```diff
@@ -461,6 +473,11 @@ def shift(self: _TDT, periods: int = 1, freq=None) -> _TDT:
         )
         return type(self)._simple_new(result, name=self.name)
 
+    @cache_readonly
+    @doc(DatetimeLikeArrayMixin.inferred_freq)
+    def inferred_freq(self) -> str | None:
```

Comment:
Should theoretically be in `DatetimeIndexOpsMixin` but it is not needed for `PeriodIndex` (see https://github.com/pandas-dev/pandas/pull/48015#discussion_r945174218)